src/test/scripts/functions/codegenalg/Algorithm_PCA.R

```diff
@@ -0,0 +1,97 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+#
+# This script performs Principal Component Analysis (PCA) on the given input data.
+#
+
+args <- commandArgs(TRUE)
+library("Matrix")
+
+A = readMM(paste(args[1], "A.mtx", sep=""));
+K = ncol(A);
+#ofmt = "CSV";
+projectData = 0;
+model = "";
+center = 0;
+scale = 0;
+output =  "/";
+
+#evec_dominant = matrix(0, 1, 1);
+
+if (model != "") {
+  # reuse existing model to project data
+} else if (model == "") {
+    model = output;
+
+  N = nrow(A);
+  D = ncol(A);
+
+  # perform z-scoring (centering and scaling)
+  if (center == 1) {
+    #cm = colMeans(A);
+    A  = A - matrix(1, nrow(A), 1) %*% colMeans(A);
+  }
+  if (scale == 1) {
+    cvars = (colSums(A^2));
+    if (center == 1){
+      #cm = colMeans(A);
+      cvars = (cvars - N*(colMeans(A)^2))/(N-1);
+    }
+    Azscored = A / sqrt(cvars);
```

Comment:
This will probably not work because R does not support matrix-vector binary operations. You can replicate cvars via a matrix multiply (e.g., `matrix(1,nrow(A),1)%*%t(cvars)`) instead